ebos/equil/initstateequil.hh

```diff
@@ -1881,24 +1884,44 @@ private:
                                 const EquilReg&   eqreg,
                                 const Grid&       grid,
                                 const PressTable& ptable,
+                                const NumericalAquifers& aquifer,
                                 PhaseSat&         psat)
     {
         using CellPos = typename PhaseSat::Position;
         using CellID  = std::remove_cv_t<std::remove_reference_t<
             decltype(std::declval<CellPos>().cell)>>;
 
-        this->cellLoop(cells, [this, &eqreg, &grid, &ptable, &psat]
+        this->cellLoop(cells, [this, &eqreg, &grid, &ptable, &aquifer, &psat]
             (const CellID                 cell,
              Details::PhaseQuantityValue& pressures,
              Details::PhaseQuantityValue& saturations,
              double&                      Rs,
              double&                      Rv) -> void
         {
-            const auto pos = CellPos {
+            auto pos = CellPos {
                 cell, UgGridHelpers::cellCenterDepth(grid, cell)
             };
 
+            const size_t global_index = UgGridHelpers::globalCell(grid)[cell];
+
+            if (aquifer.hasCell(global_index)) {
+                const auto& aqu_cells = aquifer.aquiferCells();
+                const auto& aqu_cell = aqu_cells.at(global_index);
+                pos = CellPos {
+                    cell, aqu_cell.depth
+                };
```

Comment:
It looks like we do repeated works here related to cell_depth. Maybe it is good if we can use the results from `elementCenterDepth_` from `eclproblem`.

Reply:
I plat to get https://github.com/OPM/opm-simulators/pull/2941 merged soon. That may affect you work here. It should be trivial to rebase it but just so that you are aware.